File: llm_outputs/python_programs_corrected_by_llama/find_first_in_sorted.py
def find_first_in_sorted(arr, x):
    lo = 0
    hi = len(arr) - 1  # Changed from 'hi' to 'len(arr)-1'

    while lo <= hi:
        mid = (lo + hi) // 2

        if x == arr[mid]:
            if mid == 0 or x!= arr[mid - 1]: 
                return mid
            elif arr[hi-1] < x:  
                return -1
            else:
                hi -= 1  

        elif x <= arr[mid]:
            hi = mid - 1  # Corrected this line

        else:
            lo = mid + 1

    return -1

File: llm_outputs/java_programs_corrected_by_llama/FIND_FIRST_IN_SORTED.java
package java_programs;
import java.util.*;

public class FIND_FIRST_IN_SORTED {

    public static int find_first_in_sorted(int[] arr, int x) {
        int lo = 0;
        int hi = arr.length - 1; // changed from arr.length

        while (lo <= hi) {
            int mid = (lo + hi) / 2;

            if (arr[mid] == x && ((mid == 0) || (arr[mid - 1]!= x))) { 
                return mid;
            } else if (arr[mid] < x) {
                lo = mid + 1;
            } else {
                hi = mid - 1; // changed from mid
            }
        }

        return -1;
    }
}